src/debugAdapter/goDebug.ts

```diff
@@ -835,10 +835,13 @@ class GoDebugSession extends LoggingDebugSession {
 		if (this.delve.remotePath.length === 0) {
 			return this.convertClientPathToDebugger(filePath);
 		}
+		// When the filePath has a different path separator
+		// than the local path separator (cross-compilation),
+		// the split and join logic won't work.
+		// See github.com/microsoft/vscode-go/issues/2010.
+		filePath = filePath.replace(/\/|\\/g, this.remotePathSeparator);
```

Comment:
wonder if it's doable to write some unit tests to check these path manipulation functions.